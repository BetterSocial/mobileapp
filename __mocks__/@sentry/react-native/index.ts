export const addBreadcrumb = jest.fn();
