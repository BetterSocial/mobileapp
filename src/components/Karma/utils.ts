export const renderCircleColor = (karma: number) => {
  switch (true) {
    case karma > 87.5:
      return ['#50BA57', '#C7EECA', '#50BA57'];
    case karma > 75:
      return ['#70BF21', '#B3E57F', '#70BF21'];
    case karma > 62.5:
      return ['#A0C527', '#D1EA82', '#A0C527'];
    case karma > 50:
      return ['#D3D20C', '#F3F370', '#D3D20C'];
    case karma > 37.5:
      return ['#E4B92E', '#FFE591', '#E4B92E'];
    case karma > 25:
      return ['#F4833C', '#F9BE9A', '#F4833C'];
    case karma > 12.5:
    default:
      return ['#DF001F', '#F5B0B9', '#DF001F'];
  }
};
