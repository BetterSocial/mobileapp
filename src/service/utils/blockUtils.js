import Toast from 'react-native-simple-toast';

import {blockAnonymous, blockUser} from '../../service/blocking';

const uiBlockPostAnonymous = async(postId, source, reason, message, callback) => {
    const data = {
        postId,
        source,
        reason,
        message,
        };
    let result = await blockAnonymous(data);
    if (result.code === 201) {
        callback()
        Toast.show(
            'The user was blocked successfully. \nThanks for making BetterSocial better!',
            Toast.LONG,
        );
    } else {
        Toast.show('Your report was filed & will be investigated', Toast.LONG);
    }
}

const uiBlockUser = async(postId, userId, source, reason, message, callback) => {
    const data = {
        userId,
        postId,
        source,
        reason,
        message,
    };
    let result = await blockUser(data);
    if (result.code === 200) {
        callback()
        Toast.show(
            'The user was blocked successfully. \nThanks for making BetterSocial better!',
            Toast.LONG,
        );
    } else {
        Toast.show('Your report was filed & will be investigated', Toast.LONG);
    }
}

const uiBlockDomain = async(domainId, reason, message, source, callback) => {
    const dataBlock = {
        domainId,
        reason,
        message,
        source,
    };
    const result = await blockDomain(dataBlock);
    if (result.code === 200) {
        callback();
        Toast.show(
            'The domain was blocked successfully. \nThanks for making BetterSocial better!',
            Toast.LONG,
        );
    } else {
        Toast.show('Your report was filed & will be investigated', Toast.LONG);
    }
    console.log('result block user ', result);
}


export default {
    uiBlockPostAnonymous,
    uiBlockUser,
    uiBlockDomain
}