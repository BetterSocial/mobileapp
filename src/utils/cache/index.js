import AsyncStorage from '@react-native-async-storage/async-storage';

const CACHE_NAME = 'cache';
export const saveToCache = (cacheKey, jsonData) => {
  if (jsonData && cacheKey && typeof jsonData === 'object' && typeof cacheKey === 'string') {
    AsyncStorage.getItem(CACHE_NAME, (err, result) => {
      if (result && typeof result === 'string') {
        let parseResult = JSON.parse(result);
        parseResult = { ...parseResult, [cacheKey]: jsonData };
        AsyncStorage.setItem(CACHE_NAME, JSON.stringify(parseResult));
      } else {
        const dataSave = {
          [cacheKey]: jsonData,
        };
        AsyncStorage.setItem(CACHE_NAME, JSON.stringify(dataSave));
      }
    });
  }
};

export const getSpecificCache = (cacheKey, callback) => {
  AsyncStorage.getItem(CACHE_NAME, (error, result) => {
    if (result && typeof result === 'string') {
      const newResult = JSON.parse(result);
      if (newResult[cacheKey]) {
        callback(newResult[cacheKey]);
      } else {
        callback(null);
      }
    } else {
      callback(null);
    }
  });
};

export const removeAllCache = () => {
  AsyncStorage.clear();
};
