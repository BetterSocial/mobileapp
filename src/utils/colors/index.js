// https://chir.ag/projects/name-that-color/
export const colors = {
  bondi_blue: '#00ADB5',
  white: '#ffffff',
  black: '#0C0C0C',
  emperor: '#4F4F4F',
  gray: '#69707C',
  gray1: '#9B9FA9',
  lightgrey: '#FFFFFF',
  concrete: '#F2F2F2',
  mine_shaft: '#333333',
  bunting: '#11243D',
  alto: '#E8EBED',
  porcelain: '#ecf0f1',
  silver: '#BDBDBD',
  french_pass: '#b6e4fd',
  pattens_blue: '#ddf2fe',
  blue: '#2F80ED',
  lightblue: '#55C2FF',
  blue1: '#4782D7',
  red: '#FF2E63',
  holytosca: '#003848',
  holytosca30percent: '#0038484D',
  blackgrey: '#69707C',
  redalert: '#FF2E63',
  halfBaked: '#88CDD0',
  tradewind: '#58B1B5',
  elm: '#22878B',
  blockColor: '#FF2E63',
  darkBlue: '#4782D7',
  blueSea10: '#55C2FF',
  greenDark: '#004346',
  babyBlue: '#7B9DCD',

  signed_primary: '#4782D7',
  signed_secondary: '#6295DD',
  anon_primary: '#003848',
  anon_secondary: '#154B5B',
  radical_red: '#FF3466',
  light_silver: '#D2D4DB'
};

export const hexToRgb = (hex, opacity = 1) => {
  const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
  const r = parseInt(result[1], 16);
  const g = parseInt(result[2], 16);
  const b = parseInt(result[3], 16);
  return result ? `rgba(${r},${g},${b},${opacity})` : null;
};

export const checkIsHasColor = (color) => {
  const regex = /^#[0-9a-f]{3,6}$/i;
  return regex.test(color);
};
