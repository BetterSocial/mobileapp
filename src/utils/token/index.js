import AsyncStorage from '@react-native-async-storage/async-storage';
import {KEY_ACCESS_TOKEN, KEY_REFRESH_TOKEN} from '../constants';
export const setToken = async (value) => {
  try {
    await AsyncStorage.setItem('tkn-getstream', value);
  } catch (e) {
    console.log(e);
  }
};
export const getToken = async () => {
  try {
    const value = await AsyncStorage.getItem('tkn-getstream');
    return value;
  } catch (e) {
    console.log(e);
    return null;
  }
};
export const setUserId = async (value) => {
  try {
    await AsyncStorage.setItem('userId', value);
  } catch (e) {
    console.log(e);
  }
};
export const getUserId = async () => {
  try {
    const value = await AsyncStorage.getItem('userId');
    return value;
  } catch (e) {
    console.log(e);
    return null;
  }
};

export const removeLocalStorege = async (value) => {
  try {
    await AsyncStorage.removeItem(value);
  } catch (e) {
    console.log(e);
  }
};

const getAccessToken = async () => {
  return await AsyncStorage.getItem(KEY_ACCESS_TOKEN);
};
const clearAccessToken = async () => {
  return await AsyncStorage.removeItem(KEY_ACCESS_TOKEN);
};

const setAccessToken = async (token) => {
  await AsyncStorage.setItem(KEY_ACCESS_TOKEN, token);
};

const setRefreshToken = async (value) => {
  try {
    await AsyncStorage.setItem(KEY_REFRESH_TOKEN, value);
  } catch (e) {
    console.log(e);
  }
};

const getRefreshToken = async () => {
  try {
    return await AsyncStorage.getItem(KEY_REFRESH_TOKEN);
  } catch (e) {
    console.log(e);
  }
};

export {
  getAccessToken,
  setAccessToken,
  setRefreshToken,
  getRefreshToken,
  clearAccessToken,
};
