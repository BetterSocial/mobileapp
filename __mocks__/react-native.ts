import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN