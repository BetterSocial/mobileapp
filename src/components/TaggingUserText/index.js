import * as React from 'react'
import { StyleSheet, Text } from 'react-native'

import { COLORS } from '../../utils/theme'
import { fonts } from '../../utils/fonts'
import { getUserId } from '../../utils/token'

const TaggingUserText = ({ text, navigation = null, currentTopic = null, selfUserId = null, otherId = null }) => {
    console.log('username', text);
    const username = text.replace('@', '');


    const onClick = () => {
        // comment sebentar
        if (!navigation || (currentTopic === text.replace('@', ''))) return

        getUserId().then(selfId => {
            navigation.push('OtherProfile', {
                data: {
                    user_id: selfId,
                    other_id: otherId,
                    username,
                },
            })
        })

    }

    return <Text testID='TaggingUserTextComponent' onPress={onClick} style={styles.text}>{text}</Text>
}

const styles = StyleSheet.create({
    text: {
        color: COLORS.blue,
        fontFamily: fonts.inter[500]
    }
})

export default TaggingUserText