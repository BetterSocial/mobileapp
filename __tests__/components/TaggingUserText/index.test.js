import * as React from 'react';
import renderer from 'react-test-renderer';
import {fireEvent, render} from '@testing-library/react-native';

import TaggingUserText from '../../../src/components/TaggingUserText';
import {getUserId} from '../../../src/utils/token';

jest.mock('../../../src/utils/token');

let navigation;

describe('TaggingUserText component test', () => {
  beforeEach(() => {
    navigation = {
      push: jest.fn()
    };
  });

  it('should render and match snapshot', () => {
    const toJSON = renderer.create(<TaggingUserText text="@fajarism" />).toJSON();
    expect(toJSON).toMatchSnapshot();
  });

  it('should not navigate if navigation is null', async () => {
    getUserId.mockResolvedValue('1');

    await Promise.resolve();

    const {getByTestId} = render(<TaggingUserText text="@fajarism" navigation={navigation} />);
    const component = getByTestId('TaggingUserTextComponent');
    fireEvent.press(component);

    expect(navigation.push).not.toHaveBeenCalled();
  });

  it('should navigate to OtherProfile screen', async () => {
    getUserId.mockResolvedValue('1');

    const {getByTestId} = render(<TaggingUserText text="@fajarism" navigation={navigation} />);
    const component = getByTestId('TaggingUserTextComponent');
    fireEvent.press(component);

    await Promise.resolve();

    expect(navigation.push).toHaveBeenCalledWith('OtherProfile', {
      data: {
        user_id: '1',
        other_id: null,
        username: 'fajarism'
      }
    });
  });

  it('should not navigate if currentTopic is equal to text', async () => {
    getUserId.mockResolvedValue('1');

    await Promise.resolve();

    const {getByTestId} = render(
      <TaggingUserText text="@fajarism" navigation={navigation} currentTopic={'fajarism'} />
    );
    const component = getByTestId('TaggingUserTextComponent');
    fireEvent.press(component);

    expect(navigation.push).not.toHaveBeenCalled();
  });
});
