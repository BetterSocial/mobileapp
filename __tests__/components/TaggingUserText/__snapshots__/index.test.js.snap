// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaggingUserText component test should render and match snapshot 1`] = `
<Text
  onPress={[Function]}
  style={
    Object {
      "color": "#2C67BC",
      "fontFamily": "Inter-Medium",
    }
  }
  testID="TaggingUserTextComponent"
>
  @fajarism
</Text>
`;
